src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -96,12 +99,40 @@ def gitlab_merge_request?
       end
 
       # New pull request or new merge request
+      # TODO: implement updated_pull_request? and closed_pull_request? similarly to new_pull_request?
       def new_pull_request?
         (github_pull_request? && @scm_extractor_payload[:action] == 'opened') ||
           (gitlab_merge_request? && @scm_extractor_payload[:action] == 'open')
       end
 
-      # TODO: implement updated_pull_request? and closed_pull_request? similarly to new_pull_request?
+      def add_branch_request_file(package:)
+        case @scm_extractor_payload[:scm]
+        when 'github'
+          branch_request_file = branch_request_file_github
+        when 'gitlab'
+          branch_request_file = branch_request_file_gitlab
+        end
+
+        package.save_file({ file: branch_request_file, filename: '_branch_request' })
+      end
+
+      def branch_request_file_github
+        {
+          action: @scm_extractor_payload[:action],
+          pull_request: {
+            head: {
+              repo: { full_name: @scm_extractor_payload[:full_name] },
```

Comment:
```suggestion
              repo: { full_name: @scm_extractor_payload[:repository_full_name] },
```
Nitpick: `@scm_extractor_payload` has a couple of keys more. Here it is clear what `:full_name` refers to, but I find it more obvious to change that key to `:repository_full_name`.

Reply:
Yes you are right, will change it

Reply:
@eduardoj done